pkg/operators/localmanager/localmanager.go

```diff
@@ -417,6 +426,160 @@ func (l *localManagerTrace) EnrichEvent(ev any) error {
 	return nil
 }
 
+func (l *LocalManager) InstantiateDataOperator(ctx operators.GadgetContext) (operators.DataOperatorInstance, error) {
+	traceInstance := &localManagerTrace{
+		manager:            l,
+		enrichEvents:       false,
+		attachedContainers: make(map[*containercollection.Container]struct{}),
+		params:             l.ParamDescs().ToParams(), // TODO
+		gadgetCtx:          ctx,
+
+		eventWrappers: make(map[datasource.DataSource]*compat.EventWrapperBase),
+	}
+
+	if l.igManager == nil {
+		traceInstance.enrichEvents = false
+	}
+
+	// hack - this makes it possible to use the Attacher interface
+	var ok bool
+	traceInstance.gadgetInstance, ok = ctx.GetVar("ebpfInstance")
+	if !ok {
+		return nil, fmt.Errorf("getting ebpfInstance")
+	}
+
+	activate := true // TODO: change to false once snapshot_socket is not skipped anymore
+
+	// Check, whether the gadget requested a map from us
+	if t, ok := ctx.GetVar(gadgets.MntNsFilterMapName); ok {
+		if _, ok := t.(*ebpf.Map); ok {
+			ctx.Logger().Debugf("gadget requested map %s", gadgets.MntNsFilterMapName)
+			activate = true
+		}
+	}
+
+	// Check for data sources containing refererences to mntns/netns that we could enrich data for
+	for _, ds := range ctx.GetDataSources() {
+		mntnsFields := ds.GetFieldsWithTag("type:gadget_mntns_id")               // TODO: const
+		netnsFields := ds.GetFieldsWithTag("type:gadget_netns_id", "name:netns") // TODO: const
+		if len(mntnsFields) == 0 && len(netnsFields) == 0 {
+			continue
+		}
+
+		ctx.Logger().Debugf("ds found: %q", ds.Name())
+
+		activate = true
+
+		var err error
+
+		var mntnsField datasource.FieldAccessor
+		var netnsField datasource.FieldAccessor
+
+		for _, f := range mntnsFields {
+			ctx.Logger().Debugf("using mntns enrichment")
+			mntnsField = f
+			mntnsField.SetHidden(true)
+			// We only support one of those per DataSource for now
+			break
+		}
+		for _, f := range netnsFields {
+			ctx.Logger().Debugf("using netns enrichment")
+			netnsField = f
+			netnsField.SetHidden(true)
+			// We only support one of those per DataSource for now
```

Comment:
Some ideas are:

* you can choose one DataSource for the standard output
* you can log multiple DataSources to different files/formats
* we adopt the ncurses like experiment from earlier so you can switch outputs / show multiple datasources simultaneously on different panes
* for web interfaces (Headlamp), handling multiple sources should be just fine (tabs or something like that)

Reply:
I think there is a small misunderstanding here. This comment indicates that only one mount ns or net ns is supported by datasource (That I think makes sense because otherwise what would happen if multiple mount ns provide different values?) 

The logic to support multiple DataSources is almost all in place. IIUC we're only missing an exporter that is able to print then.

Reply:
Ah you're right, sorry - I just saw the comment on the last line and thought this was referring to the output to the CLI.

Regarding multiple netns/mntns: I don't have an idea where this would make sense (but I guess there are use cases). But it should be fairly easy to add behavior that would add container/kubernetes information on subfields rather than on the main struct if there are multiple occurrences - should we decide to support that.